pkg/resolution/resolver/git/repository.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2025 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"context"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+type repository struct {
+	url       string
+	username  string
+	password  string
+	directory string
+}
+
+func clone(ctx context.Context, url, username, password string) (*repository, func(), error) {
+	urlParts := strings.Split(url, "/")
+	repoName := urlParts[len(urlParts)-1]
+	tmpDir, err := os.MkdirTemp("", repoName+"-*")
+	if err != nil {
+		return nil, func() {}, err
+	}
+	cleanupFunc := func() {
+		os.RemoveAll(tmpDir)
+	}
+
+	repo := repository{
+		url:       url,
+		username:  username,
+		password:  password,
+		directory: tmpDir,
+	}
+
+	_, err = repo.execGit(ctx, "clone", repo.url, tmpDir, "--depth=1", "--no-checkout")
+	if err != nil {
+		if strings.Contains(err.Error(), "unable to get password from user") {
```

Comment:
can't you rely on unix exit code instead? instead trying to get the strings matches ? not sure if that apply but maybe that can get output in some other way? (ie: translated strings)

but relying on the exit code should be better:

![image](https://github.com/user-attachments/assets/d24b2e2a-99b2-4ed3-9281-aabc93def93e)

Reply:
I'd love to do something cleaner here. The exit code isn't specific to auth errors however. 
![image](https://github.com/user-attachments/assets/23b8585e-cf82-4660-8147-4b61b76334b7)

My intention was to 1) maintain mostly similar error messaging to the old implementation 2) without introducing red herrings. I didn't see another way to ensure that auth issues (or private repos, and by proxy non-existent repos) would maintain their `authentication required` error without errors like `unable to look up $domain` being surpressed. Do you have any thoughts?